plugins/urls/__init__.py

```diff
@@ -175,113 +238,74 @@ def message_handler(self, event=MessageReceived):
-                                              "tinyurl"}
-                            }
-                    if target.name not in self.channels[protocol.name]:
-                        with self.channels:
-                            self.channels[protocol.name][target.name] = {
-                                "last": url,
-                                "status": "on",
-                                "shortener":
-                                "tinyurl"
-                            }
-                    else:
-                        with self.channels:
-                            self.channels[protocol.name][target.name]["last"] \
-                                = url
-                    if title is None:
-                        return
-
-                    if domain is not None and "/" in domain:
-                        domain = domain.split("/")[0]
-                    if domain is None:
-                        target.respond(title)
-                    else:
-                        target.respond("\"%s\" at %s" % (title, domain))
-                elif isinstance(target, User):
-                    if title is None:
-                        return
+            _port = _port.lstrip(":")  # Remove this as the regex captures it
 
-                    if domain is not None and "/" in domain:
-                        domain = domain.split("/")[0]
-                    if domain is None:
-                        source.respond(title)
-                    else:
-                        source.respond("\"%s\" at %s" % (title, domain))
-                else:
-                    self.logger.warn(_("Unknown target type: %s [%s]")
-                                     % (target, target.__class__))
+            try:
+                if _port:
+                    _port = int(_port)
+            except ValueError:
+                self.logger.warn("Invalid port: {0}", _port)
+                continue
+
+            _translated = self.translate_prefix(_prefix)
+
+            if len(_path) > 0 and len(_translated) > 0:
+                # Remove translated prefix chars.
+                for char in reversed(_translated):
+                    if _path[-1] == char:
+                        _path = _path[:-1]
+            elif len(_domain) > 0 and len(_translated) > 0:
+                # Remove translated prefix chars.
+                for char in reversed(_translated):
+                    if _domain[-1] == char:
+                        _domain = _domain[:-1]
+
+            _url = URL(self, _protocol, _basic, _domain, _port, _path)
+
+            self.channels.get(protocol.name, {}) \
+                .get(source.name, {})["last"] = _url.text
+
+            yield self.run_handlers(_url, {
+                "event": event,
+                "config": self.config
+            })
 
     def urls_command(self, protocol, caller, source, command, raw_args,
-                     parsed_args):
+                     args):
```

Comment:
This no longer needs to be on a new line with its new name. I'd advise against renaming this in case we switch to calling by keyword arguments in future, but since this is in mainline, it's an easy fix if we did that anyway.